src/WCSimLIGen.cc

```diff
@@ -22,8 +22,8 @@ WCSimLIGen::WCSimLIGen(){
 WCSimLIGen::~WCSimLIGen(){
 
     // things to delete
-    delete myLIGun;
-    if(hProfile) delete hProfile;
+  //delete myLIGun;
+  //if(hProfile) delete hProfile;
```

Comment:
The answer is probably "doing this because of ROOT/Geant4 memory management complications", but just to check that you've tried running with just one delete statement and still see a seg fault at the end?